tests/robot-tests/scripts/create_snapshots.py

```diff
@@ -247,10 +292,40 @@ def create_methodologies_snapshot(self) -> None:
         token = base64.b64encode(f"{args.basic_auth_user}:{args.basic_auth_pass}".encode())
         driver.execute_cdp_cmd("Network.setExtraHTTPHeaders", {"headers": {"Authorization": f"Basic {token.decode()}"}})
 
-    snapshot_service = SnapshotService()
+    snapshot_service = SnapshotService(public_url=args.public_url, slack_webhook_url=args.slack_webhook_url)
+
+    if args.validate:
```

Comment:
I suggest we take this a little bit further and make this a bit more succinct (happy to hear if you think it's a good idea too). I think we should create a new method called `validate_all_snapshots` which will take in `enable_slack` as a boolean. We'll provide an array of snapshot paths to loop over which will then call the `validate_snapshot` function (suggest making this private if you do go down this route). If the snapshot isn't valid and there is a slack webhook URL present, we'll send a slack notification. This is simply to avoid the `if name == main` section getting too busy. With this implementation, we can simply do: 

```python
    if args.validate:
        snapshot_service.validate_snapshots(args.enable_slack_notifications)
    else:
        snapshot_service.write_all_snapshots_to_file()
```

We'll have to add one function to pass the correct `create_snapshot` function to the `validate_snapshot` function. This would be something along the lines of:

```python
    def _get_snapshot_func(self, snapshot_name: str):
        if snapshot_name == 'find_statistics_snapshot.json':
            return self.create_find_statistics_snapshot
        elif snapshot_name == 'table_tool_snapshot.json':
            return self.create_table_tool_snapshot
        elif snapshot_name == 'methodologies_snapshot.json':
            return self.create_methodologies_snapshot
        elif snapshot_name == 'data_catalogue_snapshot.json':
            return self.create_data_catalogue_snapshot
        else:
            raise Exception("Invalid snapshot name")
```
which we could then use in `validate_snapshots`: 
```python
        snapshots = [
        'find_statistics_snapshot.json',
        'table_tool_snapshot.json',
        'methodologies_snapshot.json',
        'data_catalogue_snapshot.json'
        ]
        
        for path in snapshots:
            valid = self._validate_snapshot(path, self._get_snapshot_func(path))
```

It's also worth noting that we may want to just copy the message we currently send when a snapshot is invalid such as `Find Statistics page snapshot doesn't match current page on https://explore-education-statistics.service.gov.uk/` if we decided to refactor the script this way

Reply:
Cool I've done all that and it's much neater now, almost all the methods become marked as private. Thanks for the suggestions! 🙂